Source/Core/VideoCommon/VideoConfig.h

```diff
@@ -182,6 +182,7 @@ struct VideoConfig final
   struct
   {
     APIType api_type;
+    APIType real_api_type; // the underlying API (for example, Metal if using MoltenVK)
```

Comment:
I'm not that familiar with Metal but not a huge fan of this approach.  Right now it's rather simple but it seems like it could blow out the complexity later if we have multiple apis and 'real' apis.  It also means every time we check our api, we have to question "did someone mean to check the 'real' api or did they intentionally avoid it?".

I think I'd rather pick one of two solutions:

1.  Check for api Vulkan and also do a preprocessor check for APPLE.  This would match what we're already doing in a few places.  Only negative is this doesn't work if later we add a second API that loads off Vulkan but isn't locked to a specific hardware.
2. Introduce a new "Metal" API type.  It creates a vulkan backend but the api type is passed in and that's where you load the metal library.  Then everywhere in code, we check for the Metal api.  That means a lot of updates to be done but it becomes very clear what apis are doing what logic.

Reply:
Yeah, I think the best and least invasive solution would be 1. I added the "real" type on the off chance a scenario like you suggested would happen (second API type that loads off another) but I don't think the chances of that are very high right now.